src/games/strategy/engine/framework/mapDownload/MapDownloadController.java

```diff
@@ -0,0 +1,104 @@
+package games.strategy.engine.framework.mapDownload;
+
+import java.awt.Frame;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.prefs.BackingStoreException;
+import java.util.prefs.Preferences;
+
+import javax.swing.JComponent;
+import javax.swing.JOptionPane;
+import javax.swing.SwingUtilities;
+
+import games.strategy.debug.ClientLogger;
+import games.strategy.engine.framework.GameRunner2;
+import games.strategy.util.CountDownLatchHandler;
+import games.strategy.util.EventThreadJOptionPane;
+
+
+/** Controller for in-game map download actions */
+public class MapDownloadController {
```

Comment:
Code in this class is coming from GameRunner2. It's a bit rough and could be refactored more. For this change, moving it and isolating it was of help.